### PR TITLE
Remove CLI built from C++ sources

### PR DESCRIPTION
**Pull request by polyntsov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -204,44 +204,62 @@ Prior to cloning the repository and attempting to build the project, ensure that
 To use test datasets you will need:
 - Git Large File Storage, version 3.0.2+
 
-### Building the project (first option: with tests)
+### Building the project
+#### Building the Python module using pip (this is what you probably want, if you're not a developer)
```
Comment (chernishev) [inline README.md:208]: you're -> you are

File: README.md [hunk 2]
```diff
@@ -204,44 +204,62 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
-cd Desbordante
-./pull_datasets.sh
-./build.sh
+python3 -m venv venv
+source venv/bin/activate
+python3 -m pip install .
 ```
 
-### Building the project (second option: without tests)
+Now it's possible to `import desbordante` as a module from within created virtual environment. In particular, now CLI can be used:
```
Comment (chernishev) [inline README.md:217]: 1) it's -> it is
2) In particular... -> Also, this step is mandatory for setting up the CLI interface of Desbordante.

File: README.md [hunk 3]
```diff
@@ -204,44 +204,62 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
+
+`desbordante.cpython-*.so` is a Python module, packaging Python bindings for Desbordante core library. In order to use it, simply `import` it:
 ```
-./Desbordante_run --algo=tane --data=<path_to_dataset>
+cd build/target
+python3
+>>> import desbordante
 ```
 
+We use [easyloggingpp](https://github.com/abumq/easyloggingpp) in order to log (mostly debug) information in the core library. Python bindings search for a configuration file in the directory from which `desbordante` was imported, so if you want to configure logging, create `logging.conf` in the directory you're importing desbordante form. This means, that for CLI `logging.conf` should be created in a directory from which you execute `cli.py`.
```
Comment (chernishev) [inline README.md:261]: 1) you're -> you are
2) form -> from
3) execute -> run

**Comment by BUYT-1** [comment]
[This](https://github.com/polyntsov/Desbordante/blob/593003eec62bc15e5bc86b9f56bfdc263d389c03/src/core/algorithms/algo_factory.h#L22) should also be removed

File: README.md [hunk 4]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
 - Git Large File Storage, version 3.0.2+
 
-### Building the project (first option: with tests)
+### Building the project
+#### Building the Python module using pip (this is what you probably want, if you are not a developer)
+**NOTE**: this step is mandatory for setting up the Command Line Interface for Desbordante.
+
 Firstly, navigate to a desired directory.
-Then, clone the repository, cd into the project directory and launch the build script:
+Then, clone the repository, cd into the project directory and launch the following commands:
```
Comment (BUYT-1) [inline README.md:212]:
```suggestion
Then, clone the repository, cd into the project directory and run the following commands:
```

File: README.md [hunk 5]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
 To use test datasets you will need:
 - Git Large File Storage, version 3.0.2+
 
-### Building the project (first option: with tests)
+### Building the project
+#### Building the Python module using pip (this is what you probably want, if you are not a developer)
```
Comment (BUYT-1) [inline README.md:208]:
```suggestion
#### Building the Python module using pip (this is what you probably want if you are not a developer)
```

File: README.md [hunk 6]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
-cd Desbordante
-./pull_datasets.sh
-./build.sh
+python3 -m venv venv
+source venv/bin/activate
+python3 -m pip install .
 ```
 
-### Building the project (second option: without tests)
+Now it is possible to `import desbordante` as a module from within created virtual environment. The CLI for Desbordante can be ran using the following commands:
```
Comment (BUYT-1) [inline README.md:219]:
```suggestion
Now it is possible to `import desbordante` as a module from within the created virtual environment. The CLI for Desbordante can be run using the following commands:
```

File: README.md [hunk 7]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
+```sh
+pip install -r cli/requirements.txt
+python3 cli/cli.py --help
+```
 
-Firstly, navigate to a desired directory.
-Then, clone the repository, cd into the project directory and launch the build script:
+#### Building tests & the Python module manually
+In order to build tests, firstly pull test datasets using the following command:
```
Comment (BUYT-1) [inline README.md:226]:
```suggestion
In order to build tests, pull the test datasets using the following command:
```

File: README.md [hunk 8]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
 ```
-git clone https://github.com/Mstrutov/Desbordante/
-cd Desbordante
-./build.sh --no-tests --no-unpack
+then build the tests themselves:
+```sh
+./build.sh -j$(nproc)
+```
+
+The Python module is built by providing `--pybind` switch:
```
Comment (BUYT-1) [inline README.md:235]:
```suggestion
The Python module can be built by providing `--pybind` switch:
```

File: README.md [hunk 9]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
+```sh
+./build.sh --pybind -j$(nproc)
 ```
 
-### Launching the binaries
-The script generates the following file structure in `/path/to/Desbordante/build/target`:
+For more available options take a look at `./build.sh --help`.
```
Comment (BUYT-1) [inline README.md:240]:
```suggestion
See `./build.sh --help` for more available options.
```
We are going for a formal style, right?

File: README.md [hunk 10]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
-The `input_data` directory contains several .csv files that may be used by `Desbordante_test`. Run `Desbordante_test` to perform unit testing:
+
+The `input_data` directory contains several .csv files that are used by `Desbordante_test`. Run `Desbordante_test` to perform unit testing:
 ```
 cd build/target
 ./Desbordante_test
 ```
-The tool itself may be run like the following:
+
+`desbordante.cpython-*.so` is a Python module, packaging Python bindings for Desbordante core library. In order to use it, simply `import` it:
```
Comment (BUYT-1) [inline README.md:256]:
```suggestion
`desbordante.cpython-*.so` is a Python module, packaging Python bindings for the Desbordante core library. In order to use it, simply `import` it:
```

File: README.md [hunk 11]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
 ```
-./Desbordante_run --algo=tane --data=<path_to_dataset>
+cd build/target
+python3
+>>> import desbordante
 ```
 
+We use [easyloggingpp](https://github.com/abumq/easyloggingpp) in order to log (mostly debug) information in the core library. Python bindings search for a configuration file in the directory from which `desbordante` was imported, so if you want to configure logging, create `logging.conf` in the directory you are importing desbordante from. This means, that for CLI `logging.conf` should be created in a directory from which you run `cli.py`.
```
Comment (BUYT-1) [inline README.md:263]:
```suggestion
We use [easyloggingpp](https://github.com/abumq/easyloggingpp) in order to log (mostly debug) information in the core library. Python bindings search for a configuration file in the directory from which `desbordante` was imported, so to configure logging, create `logging.conf` in the directory desbordante will be imported from. This means that for CLI `logging.conf` should be created in the directory from which `cli.py` is run.
```
"For CLI"?

File: README.md [hunk 12]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
 - Git Large File Storage, version 3.0.2+
 
-### Building the project (first option: with tests)
+### Building the project
+#### Building the Python module using pip (this is what you probably want if you are not a developer)
+**NOTE**: this step is mandatory for setting up the Command Line Interface for Desbordante.
+
 Firstly, navigate to a desired directory.
-Then, clone the repository, cd into the project directory and launch the build script:
+Then, clone the repository, cd into the project directory and run the following commands:
```
Comment (BUYT-1) [inline README.md:212]: nit:
```suggestion
Then, clone the repository, `cd` into the project directory and run the following commands:
```

File: README.md [hunk 13]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
 ```
-git clone https://github.com/Mstrutov/Desbordante/
-cd Desbordante
-./build.sh --no-tests --no-unpack
+then build the tests themselves:
+```sh
+./build.sh -j$(nproc)
+```
+
+The Python module can be built by providing `--pybind` switch:
```
Comment (BUYT-1) [inline README.md:235]:
```suggestion
The Python module can be built by providing the `--pybind` switch:
```

File: README.md [hunk 14]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
+
+The `input_data` directory contains several .csv files that are used by `Desbordante_test`. Run `Desbordante_test` to perform unit testing:
 ```
 cd build/target
 ./Desbordante_test
 ```
-The tool itself may be run like the following:
+
+`desbordante.cpython-*.so` is a Python module, packaging Python bindings for the Desbordante core library. In order to use it, simply `import` it:
 ```
```
Comment (BUYT-1) [inline README.md:257]:
````suggestion
```sh
````

File: README.md [hunk 15]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
-./Desbordante_run --algo=tane --data=<path_to_dataset>
+cd build/target
+python3
+>>> import desbordante
 ```
 
+We use [easyloggingpp](https://github.com/abumq/easyloggingpp) in order to log (mostly debug) information in the core library. Python bindings search for a configuration file in the directory from which `desbordante` was imported, so to configure logging, create `logging.conf` in the directory desbordante will be imported from. This means that when running CLI `logging.conf` should be created in the directory from which `cli.py` is run.
```
Comment (BUYT-1) [inline README.md:263]:
```suggestion
We use [easyloggingpp](https://github.com/abumq/easyloggingpp) in order to log (mostly debug) information in the core library. Python bindings search for a configuration file in the working directory, so to configure logging, create `logging.conf` in the directory from which desbordante will be imported. In particular, when running the CLI, `logging.conf` should be located in the directory from which the command to start it is to be issued.
```

File: README.md [hunk 16]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
 To use test datasets you will need:
 - Git Large File Storage, version 3.0.2+
 
-### Building the project (first option: with tests)
+### Building the project
+#### Building the Python module using pip (this is what you probably want if you are not a developer)
+**NOTE**: this step is mandatory for setting up the Command Line Interface for Desbordante.
+
 Firstly, navigate to a desired directory.
```
Comment (BUYT-1) [inline README.md:211]:
```suggestion
Firstly, navigate to your directory of choice.
```

File: README.md [hunk 17]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
+```sh
+./build.sh --pybind -j$(nproc)
 ```
 
-### Launching the binaries
-The script generates the following file structure in `/path/to/Desbordante/build/target`:
+See `./build.sh --help` for more available options.
+
+The `./build.sh` script generates the following file structure in `/path/to/Desbordante/build/target`:
 ```bash
```
Comment (BUYT-1) [inline README.md:243]:
````suggestion
```
````

**Review by BUYT-1 — CHANGES_REQUESTED** [review]

File: README.md (hunk 12 above)
Comment (BUYT-1) [inline README.md:212]:
```suggestion
Then, clone the repository, change the current directory to the project directory and run the following commands:
```

File: README.md (hunk 15 above)
Comment (polyntsov) [reply README.md:263]: > the command to start it is to be issued.

I think it's better to be specific here

File: README.md (hunk 16 above)
Comment (BUYT-1) [reply README.md:211]: You're free to reject it, but "of choice" is more common. Search "to a directory of choice" and "to a desired directory" in Google and the numbers will be significantly different.

Comment (polyntsov) [reply README.md:211]: OK, I agree, good point.

File: README.md (hunk 15 above)
Comment (BUYT-1) [reply README.md:263]: Oh, is it expected to install `cli.py` to some directory in `$PATH`?

Comment (BUYT-1) [reply README.md:263]: Keeping the name `cli.py` is a pretty bad idea then.

Comment (BUYT-1) [reply README.md]: > > the command to start it is to be issued.
> 
> I think it's better to be specific here

The specific command is going to look like `/path/to/Desbordante/cli/cli.py` or `python3 /path/to/Desbordante/cli/cli.py` if it's not in `$PATH`

Comment (BUYT-1) [reply README.md]: > > Keeping the name cli.py is a pretty bad idea then.
> 
> why?

I added it a few seconds after. If it's in `$PATH`, it should have a distinct name. Any number of programs could be called `cli.py`, and there should be no ambiguity here.

Comment (BUYT-1) [reply README.md]:
```suggestion
We use [easyloggingpp](https://github.com/abumq/easyloggingpp) in order to log (mostly debug) information in the core library. Python bindings search for a configuration file in the working directory, so to configure logging, create `logging.conf` in the directory from which desbordante will be imported. In particular, when running `python3 ./relative/path/to/cli.py`, `logging.conf` should be located in `.`.
```

File: README.md [hunk 18]
```diff
@@ -204,44 +204,64 @@ Prior to cloning the repository and attempting to build the project, ensure that
...
 │   └───some-sample-csv\'s.csv
 ├───Desbordante_test
-├───Desbordante_run
-```
-The `input_data` directory contains several .csv files that may be used by `Desbordante_test`. Run `Desbordante_test` to perform unit testing:
+├───desbordante.cpython-*.so
 ```
+
+The `input_data` directory contains several .csv files that are used by `Desbordante_test`. Run `Desbordante_test` to perform unit testing:
+```bash
```
Comment (BUYT-1) [inline README.md:251]: nit:
````suggestion
```sh
````

**Review by BUYT-1 — APPROVED** [review]

File: README.md (hunk 15 above)
Comment (BUYT-1) [reply README.md]: Oh, "`cli.py`" isn't the exact command, it's the file, right? Then the current version is fine, though some further emphasis would be nice ("is to be run." -> "is to be run, which is not necessarily the same directory it is located in.", if that's what is meant).

**Review by chernishev — APPROVED** [review]

**Pull request merged**